rustls/src/quic.rs

```diff
@@ -679,6 +679,20 @@ pub trait PacketKey: Send + Sync {
 
     /// Tag length for the underlying AEAD algorithm
     fn tag_len(&self) -> usize;
+
+    /// Number of messages that can be safely encrypted with a single key of this type.
+    ///
+    /// See [`CipherSuiteCommon::confidentiality_limit`][csc-limit].
+    ///
+    /// [csc-limit]: crate::crypto::CipherSuiteCommon::confidentiality_limit
+    fn confidentiality_limit(&self) -> u64;
```

Comment:
Agreed, wouldn't this fall under the category of [Major: adding a non-defaulted trait item](https://doc.rust-lang.org/cargo/reference/semver.html#trait-new-item-no-default) ?

Are we going to swallow this semver break on the assumption it's rare outside of Quinn to have this trait implemented?

Reply:
> Are we going to swallow this semver break on the assumption it's rare outside of Quinn to have this trait implemented?

I think since this is slotted for https://github.com/rustls/rustls/issues/1777 it isn't a concern. I assume Quinn will jump over 0.22.

Reply:
Right. I think we might want to reconsider how `quic::Algorithm` is instantiated, though, to avoid the duplication identified in the other comment thread?